Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  Alicerce.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public final class Log: Logger {
+
+    public final class Queue {
+        public let dispatchQueue: DispatchQueue
+
+        public init(label: String, qos: DispatchQoS = .background) {
+            self.dispatchQueue = DispatchQueue(label: label, qos: qos)
+        }
+    }
+
+    public private(set) var destinations = [LogDestination]()
+
+    public static let defaultLevel = Level.error
+    public var errorClosure: ((LogDestination, Log.Item, Error) -> ())?
+
+    // MARK:- Destination Management
+
+    public func register(_ destination: LogDestination) {
+
+        if destinations.contains(where: { $0.instanceId == destination.instanceId }) == false {
```

Comment:
Accessing `destinations` is not thread safe, so we should synchronise access to it, e.g. using a serial queue.

This applies to essentially all the public API's, and while on the register/unregister its not so critical (we normally set this up on startup in the main thread), the log API's will also not be thread safe and message order may also not be respected. We are also not defending against possible `LogDestinations` not being async.